src/stages/sender_recovery.rs

```diff
@@ -0,0 +1,300 @@
+use crate::{
+    accessors::chain,
+    stagedsync::stage::{ExecOutput, Stage, StageInput, SyncError, UnwindInput},
+    MutableTransaction, StageId,
+};
+use async_trait::async_trait;
+use ethereum::{Transaction, TransactionMessage};
+use ethereum_types::Address;
+use secp256k1::{
+    recovery::{RecoverableSignature, RecoveryId},
+    Message, SECP256K1,
+};
+use sha3::{Digest, Keccak256};
+use std::cmp;
+use tracing::error;
+
+const BUFFER_SIZE: u64 = 5000;
+
+fn recover_sender(tx: &Transaction) -> anyhow::Result<Address> {
+    let mut sig = [0u8; 64];
+    sig[..32].copy_from_slice(tx.signature.r().as_bytes());
+    sig[32..].copy_from_slice(tx.signature.s().as_bytes());
+
+    let rec = RecoveryId::from_i32(tx.signature.standard_v() as i32)?;
+
+    let public = &SECP256K1.recover(
+        &Message::from_slice(TransactionMessage::from(tx.clone()).hash().as_bytes())?,
+        &RecoverableSignature::from_compact(&sig, rec)?,
+    )?;
+
+    let address_slice = &Keccak256::digest(&public.serialize_uncompressed()[1..])[12..];
+    Ok(Address::from_slice(address_slice))
+}
+
+async fn process_block<'tx, RwTx>(tx: &mut RwTx, height: u64) -> anyhow::Result<()>
+where
+    RwTx: MutableTransaction<'tx>,
```

Comment:
```suggestion
async fn process_block<'db: 'tx, 'tx, RwTx>(tx: &'tx mut RwTx, height: u64) -> anyhow::Result<()>
where
    RwTx: MutableTransaction<'db>,
```

Maybe stick to original lifetimes, if only for readability?